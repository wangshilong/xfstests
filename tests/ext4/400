#! /bin/bash
# FS QA Test No. 400
#
# Test to verify that turn group/project quotas off while user quotas
# are left on.
#
#-----------------------------------------------------------------------
# Copyright (c) 2013 Oracle, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/quota

# Modify as appropriate.
_supported_fs ext4
_supported_os Linux

_require_scratch

_scratch_mkfs_ext4 -O quota,project >> $seqres.full 2>&1
_scratch_mount >> $seqres.full 2>&1

QUOTA_DIR=$SCRATCH_MNT/quota_dir
quotaon -ugP $SCRATCH_MNT

mkdir -p $QUOTA_DIR
echo "*** turn off group quotas"
quotaoff -g $SCRATCH_MNT
rmdir $QUOTA_DIR
echo "*** umount"
umount $SCRATCH_MNT

_scratch_mount >> $seqres.full 2>&1
quotaon -ugP $SCRATCH_MNT
mkdir -p $QUOTA_DIR
echo "*** turn off project quotas"
quotaoff -P $SCRATCH_MNT
rmdir $QUOTA_DIR
echo "*** umount"
umount $SCRATCH_MNT

_scratch_mount >> $seqres.full 2>&1
quotaon -ugP $SCRATCH_MNT
mkdir -p $QUOTA_DIR
echo "*** turn off group/project quotas"
quotaoff -gP $SCRATCH_MNT
rmdir $QUOTA_DIR
echo "*** umount"
umount $SCRATCH_MNT

# success, all done
status=0
exit
