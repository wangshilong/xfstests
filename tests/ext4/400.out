QA output created by 400
*** turn off group quotas
*** umount
*** turn off project quotas
*** umount
*** turn off group/project quotas
*** umount
